tap_mysql/client.py

```diff
@@ -156,6 +156,42 @@ def sdk_typing_object(
 
         return sqltype_lookup["string"]  # safe failover to str
 
+    def discover_catalog_entries(self) -> list[dict]:
+        """Return a list of catalog entries from discovery.
+
+        Override to only process schemas specified in config.
+
+        Returns:
+            The discovered catalog entries as a list.
+        """
+        result: list[dict] = []
+        engine = self._engine
+        inspected = sqlalchemy.inspect(engine)
+        for schema_name in self.get_schema_names(engine, inspected):
```

Comment:
Why not implement this like the tap-postgres implementation? Seemed cleaner than this  https://github.com/MeltanoLabs/tap-postgres/pull/218/files and seems like we could use it in the SDK as well. If we just did that it'd bypass the need for the rest of this right?